packages/@aws-cdk/aws-stepfunctions-tasks/lib/eks/create-nodegroup.ts

```diff
@@ -0,0 +1,304 @@
+import * as iam from '@aws-cdk/aws-iam';
+import * as sfn from '@aws-cdk/aws-stepfunctions';
+import { Construct } from 'constructs';
+import { integrationResourceArn, validatePatternSupported } from '../private/task-utils';
+
+/** Properties for creating a Nodegroup with EksCreateNodegroup */
+export interface EksCreateNodegroupProps extends sfn.TaskStateBaseProps {
+
+  /** The name of the cluster to create the node group in */
+  readonly clusterName: string;
+
+  /** The unique name to give your node group */
+  readonly nodegroupName: string;
+
+  /**
+   * The scaling configuration details for the Auto Scaling group that is created for your node group
+   *
+   * @default - no scaling config
+   */
+  readonly scalingConfig?: NodeGroupScalingConfig;
+
+  /**
+   * The root device disk size (in GiB) for your node group instances
+   *
+   * @default - 20 GiB
+   */
+  readonly diskSize?: number;
+
+  /**
+   * The subnets to use for the Auto Scaling group that is created for your node group
+   * @example subnets must have the tag key kubernetes.io/cluster/CLUSTER_NAME with a value of shared , where CLUSTER_NAME is replaced with the name of your cluster
+   */
+  readonly subnets: string[];
+
+  /**
+   * The instance type to use for your node group
+   *
+   * @default - t3.medium
+   */
+  readonly instanceTypes?: string[];
+
+  /**
+   * AMI type for your node group
+   *
+   * @default - GPU uses AL2_x86_64_GPU, Non-GPU uses AL2_x86_64, Arm uses AL2_ARM_64
+   */
+  readonly amiType?: string;
+
+  /**
+   * The remote access (SSH) configuration to use with your node group
+   * you specify launchTemplate , then don't specify remoteAccess , or the node group deployment will fail
+   *
+   * @default - no remote access config
+   */
+  readonly remoteAccess?: RemoteAccessConfig;
+
+  /** The Amazon Resource Name (ARN) of the IAM role to associate with your node group */
+  readonly nodeRole: string;
+
+  /**
+   * The Kubernetes labels to be applied to the nodes in the node group when they are created
+   *
+   * @default - no labels
+   */
+  readonly labels?: Record<string, string>;
```

Comment:
have you run `yarn package` - does `jsii` complain about using `Record` here?
i'm not entirely certain if this can generate the clients for all lanauges